azurerm/internal/services/policy/parse/definition.go

```diff
@@ -0,0 +1,43 @@
+package parse
+
+import (
+	"fmt"
+	"regexp"
+)
+
+type PolicyDefinitionId struct {
+	Name string
+	PolicyScopeId
+}
+
+// TODO: This paring function is currently suppressing every case difference due to github issue: https://github.com/Azure/azure-rest-api-specs/issues/8353
+func PolicyDefinitionID(input string) (*PolicyDefinitionId, error) {
+	// in general, the id of a definition should be:
+	// {scope}/providers/Microsoft.PolicyInsights/remediations/{name}
```

Comment:
Yeah... This regex only matches the middle part, and then I did a split on the ID to separate the `{scope}` part and the `name`. Because the `{scope}` can either be a subscription ID, resource group ID, resource ID, or management group ID, those 4 scopes do not share the same pattern.

Reply:
kk 👍